### PR TITLE
add ability to hook into any click()/get() method and wait for page l…

### PR DESCRIPTION
**Pull request by hmcgowan**
Sample impl:

```
DriverManager.addPageLoadWaiter(new PageLoadWaiter() {
            @Override
            public TimeUnit getTimeUnit() {
                return TimeUnit.SECONDS;
            }

            @Override
            public int getTimeout() {
                return 2;
            }

            @Override
            public boolean isSatisfied() {
                log.info("do some wait here");
                return true;
            }
        });
```

### CODE REVIEW

File: src/main/java/com/lithium/mineraloil/selenium/browsers/PageLoadWaiter.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.lithium.mineraloil.selenium.browsers;
+
+public interface PageLoadWaiter {
+    void execute();
```
Comment (jchang691) [inline src/main/java/com/lithium/mineraloil/selenium/browsers/PageLoadWaiter.java]: Can we name this differently. Maybe wait()?

Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/browsers/PageLoadWaiter.java]: wait() is an Object method....

Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/browsers/PageLoadWaiter.java]: I'm wondering if it should take a WaitCondition and pass an interval and timeout

Comment (jchang691) [inline src/main/java/com/lithium/mineraloil/selenium/browsers/PageLoadWaiter.java]: that could work

File: src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java [hunk 2]
```diff
@@ -238,4 +243,20 @@ public static File takeScreenshot() {
...
+        for (PageLoadWaiter pageLoadWaiter : pageLoadWaiters) {
+            new WaitCondition() {
+                @Override
+                public boolean isSatisfied() {
+                    return pageLoadWaiter.isSatisfied();
+                }
+            }.setTimeout(pageLoadWaiter.getTimeUnit(), pageLoadWaiter.getTimeout())
+             .throwExceptionOnFailure(new PageLoadWaiterTimeoutException("Failure running custom page load waiter"))
+             .waitUntilSatisfied();
+        }
```
Comment (jchang691) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: If a timeout occurs for a certain pageLoadWaiter, will it be easy to track down which pageLoadWaiter failed if there are multiple? The stack trace will just point to here right?

Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: great comment. thinking about this

Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: How's this message?
com.lithium.mineraloil.selenium.exceptions.PageLoadWaiterTimeoutException: Timed out in PageLoadWaiter: package 'lsw.junit.rules', class 'lsw.junit.rules.OnUIStart'

Comment (hmcgowan) [inline src/main/java/com/lithium/mineraloil/selenium/elements/DriverManager.java]: should show the class and package even though the trace is in mineraloil-selenium

**Pull request merged**